opal/core/search/search_query.py

```diff
@@ -0,0 +1,114 @@
+from opal.core.discoverable import DiscoverableFeature
+from opal.utils import camelcase_to_underscore
+from opal.models import Episode, deserialize_date
+
+
+class SearchException(Exception):
+    pass
+
+
+class SearchQueryField(object):
+    lookup_list = None
+    enum = None
+    description = None
+    slug = None
+    display_name = None
+
+    def get_slug(self):
+        if self.slug is not None:
+            return self.slug
+
+        if self.display_name is None:
+            raise ValueError(
+                'Must set display_name for {0}'.format(self)
+            )
+        return camelcase_to_underscore(self.display_name).replace(' ', '')
+
+    def get_display_name(self):
+        return self.display_name
+
+    def to_dict(self):
+        return dict(
+            name=self.get_slug(),
+            title=self.display_name,
+            type=self.field_type,
+            enum=self.enum,
+            lookup_list=self.lookup_list,
+            description=self.description
+        )
+
+    def query(self, given_query):
+        """
+            takes in the full query and returns a list of episodes
+        """
+        raise NotImplementedError("please implement a query")
+
+
+class SearchQuery(DiscoverableFeature):
+    module_name = "search_query"
+    fields = []
+
+    def get_fields(self):
+        return self.fields
+
+    def to_dict(self):
+        return dict(
+            name=self.get_slug(),
+            display_name=self.display_name,
+            advanced_searchable=True,
+            fields=[i().to_dict() for i in self.get_fields()]
+        )
```

Comment:
I don't really see why you need to instantiate these... particularly given it means they then work differently to the discoverable methods ?